app/src/main/java/com/example/pygmyhippo/organizer/ProfileFragment.java

```diff
@@ -85,15 +130,158 @@ public View onCreateView(@NonNull LayoutInflater inflater,
+
+                // remove focus in touch mode
+                name_f.setFocusableInTouchMode(false);
+                pronoun_f.setFocusableInTouchMode(false);
+                phone_f.setFocusableInTouchMode(false);
+                email_f.setFocusableInTouchMode(false);
+                facilityName_f.setFocusableInTouchMode(false);
+                facilityLocation_f.setFocusableInTouchMode(false);
+
+                // set visibility
+                updateButton.setVisibility(View.GONE);
+                editButton.setVisibility(View.VISIBLE);
+                uploadIm_btn.setVisibility(View.GONE);
+                deleteIm_btn.setVisibility(View.GONE);
+                facility_uploadIm_btn.setVisibility(View.GONE);
+
+            }
+        };
+
+
+        // Listener for the uploadAvatar image button, it allows the user to select a photo from their photo gallery by launching the media picker
+        View.OnClickListener uploadAvatar = new View.OnClickListener() {
+            /**
+             * Tells the media picker to launch when the button listener is triggered
+             * @author Jennifer
+             * @param view the fragment view
+             */
+            @Override
+            public void onClick(View view) {
+                uploadType = "avatar";
+                // Launch the photo picker and let the user choose only images.
+                pickMedia.launch(new PickVisualMediaRequest.Builder()
+                        .setMediaType(ActivityResultContracts.PickVisualMedia.ImageOnly.INSTANCE)
+                        .build());
+            }
+        };
+
+        View.OnClickListener uploadFacility = new View.OnClickListener() {
+            /**
+             * Tells the media picker to launch when the button listener is triggered
+             * @author Jennifer
+             * @param view the fragment view
+             */
+            @Override
+            public void onClick(View view) {
+                uploadType = "facility";
+                // Launch the photo picker and let the user choose only images.
+                pickMedia.launch(new PickVisualMediaRequest.Builder()
+                        .setMediaType(ActivityResultContracts.PickVisualMedia.ImageOnly.INSTANCE)
+                        .build());
+            }
+        };
+
+
+        // On click Listener for the delete image button
+        View.OnClickListener delete = new View.OnClickListener() {
+            /**
+             * Sends the users name to the method getAvatar
+             * @author Jennifer
+             * @param view the fragment view
+             */
+            @Override
+            public void onClick(View view) {
+                try {
+                    getAvatar(name_f.getText().toString());
+                } catch (URISyntaxException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        };
+
+        facility_uploadIm_btn.setOnClickListener(uploadFacility);
```

Comment:
What do setting these onClickListeners do? Do they like generally call on click cause that would be handy

Reply:
This will run the code that opens the photopicker

Reply:
The profile picture will change depending on what you enter into the name field. The automatic avatar generation is mostly until I place minimum length requirements on the name field. Currently it will create something without you entering anything.

Reply:
Oh so does the name need to be "Updated" before you can see the relation?

Reply:
Yes, updating the name will force a new avatar to be created. If the name box is empty it generates for a string that says null. Either the program or the website caches the avatar so it will not generate a new one with the same name.

Reply:
So same names will have the same avatar?

Reply:
If there are exactly the same then yes, but this does fit the user story. 

"As an entrant I want my profile picture to be deterministically generated from my profile name if I haven't uploaded a profile image yet."